### PR TITLE
Kubernetes NetworkPolicy fixes & enhancements

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Review by aanm — APPROVED** [review]

File: pkg/k8s/network_policy.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	for _, iRule := range np.Spec.Ingress {
+		// Based on NetworkPolicyIngressRule docs:
+		//   From []NetworkPolicyPeer
+		//   If this field is empty or missing, this rule matches all
+		//   sources (traffic not restricted by source).
+		if iRule.From == nil || len(iRule.From) == 0 {
+			all := labels.NewLabel(labels.IDNameAll, "", common.ReservedLabelSource)
+			ar := &policy.AllowRule{
+				Action: api.ALWAYS_ACCEPT,
+				Labels: labels.LabelArray{all},
```
Comment (ianvernon) [inline pkg/k8s/network_policy.go]: This means default allow traffic?

File: pkg/policy/api/consumable_decision.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import (
+	"fmt"
+)
+
+// ConsumableDecision is a decision when deciding ingress/egress consuming
+type ConsumableDecision byte
+
+const (
+	// UNDECIDED means unsed while the decision is still undecided
```
Comment (ianvernon) [inline pkg/policy/api/consumable_decision.go:25]: unsed? undecided about what? please clarify in the comment.

Comment (tgraf) [reply pkg/policy/api/consumable_decision.go:25]: Undecided means that the policy decision has not been made yet. I haven't introduced this code in this commit, I've just moved it into a separate package.

Comment (ianvernon) [reply pkg/policy/api/consumable_decision.go:25]: Noted, thanks.

**Review by ianvernon — APPROVED** [review]
Core changes look good, as does the refactoring.

**Review by aalemayhu — COMMENTED** [review]
lgtm

**Pull request merged**